### ISSUE TITLE
Upgrade Quill to remove the deprecated DOMNodeInserted mutation event

### ISSUE THREAD

**Issue by lindalinya** [zenoamaro/react-quill#970]
The quill has released 2.0.0 at [quill-release](https://github.com/quilljs/quill/releases/tag/v2.0.0)
In the old quill version, some deprecated events are used in the code.  [mutation-events-deprecation](https://developer.chrome.com/blog/mutation-events-deprecation)
quill 2.0 has removed that event from the v2.0.0 [#3806](https://github.com/quilljs/quill/issues/3806)
When will react-quill update the quill version?

**Comment by JSchmidt63** [comment]
[Mutation events will be removed from Chrome](https://developer.chrome.com/blog/mutation-events-deprecation)
there is Polyfill, seen it at
[mutation-events Polyfill](https://github.com/mfreed7/mutation-events-polyfill#readme)

**Comment by MubashirullahD** [comment]
### High priority issue
This could be a breaking problem and if not addressed would require us to migrate to a different solution.


> [Deprecation] Listener added for a 'DOMNodeInserted' mutation event. This event type is deprecated, and will be removed from this browser very soon. Usage of this event listener will cause performance issues today, and represents a large risk of future site breakage. Consider using MutationObserver instead. See https://chromestatus.com/feature/5083947249172480 for more information.

**Comment by lindalinya** [comment]
In the blink of an eye, it's July, and only 30 days have passed since Chrome made these official removals. There is still no official answer to this issue. Perhaps developers should start choosing alternatives. I wonder if there are any open source alternatives other than Polyfill

**Comment by hugopqt** [comment]
Yes so same issue here, I use "react-quill": "^2.0.0" in my react app, and in the browser I receive "quill.js:4233 [Deprecation] Listener added for a 'DOMNodeInserted' mutation event. This event type is deprecated, and will be removed from this browser VERY soon. Usage of this event listener will cause performance issues today, and represents a large risk of imminent site breakage. Consider using MutationObserver instead. See https://chromestatus.com/feature/5083947249172480 for more information." I was hoping it would be fix it in 2.0.0 but still I receive this warning, knowing that there wont be mutation events in Chrome 127 (going stable on July 30 2024) it seems like a big issue for us, any fix coming in the next weeks?

**Comment by jd-frc** [comment]
I'm getting a DOM mutation event deprecation warning with 2.0.2, freshly downloaded today:
```
recipeDetail.js:170 [Violation] Listener added for a 'DOMNodeInserted' mutation event. This event type is deprecated, and will be removed from this browser VERY soon. Usage of this event listener will cause performance issues today, and represents a large risk of imminent site breakage. Consider using MutationObserver instead. See https://chromestatus.com/feature/5083947249172480 for more information.
Scroll	@	quill.js:4237
create	@	quill.js:184
Quill	@	quill.js:1153
ReactQuill.createEditor	@	index.tsx:332
ReactQuill.instantiateEditor	@	index.tsx:290
ReactQuill.componentDidMount	@	index.tsx:253
commitLifeCycles	@	react-dom.development.js:20663
commitLayoutEffects	@	react-dom.development.js:23426
callCallback	@	react-dom.development.js:3945
invokeGuardedCallbackDev	@	react-dom.development.js:3994
invokeGuardedCallback	@	react-dom.development.js:4056
commitRootImpl	@	react-dom.development.js:23151
unstable_runWithPriority	@	scheduler.development.js:468
runWithPriority$1	@	react-dom.development.js:11276
commitRoot	@	react-dom.development.js:22990
performSyncWorkOnRoot	@	react-dom.development.js:22329
(anonymous)	@	react-dom.development.js:11327
unstable_runWithPriority	@	scheduler.development.js:468
runWithPriority$1	@	react-dom.development.js:11276
flushSyncCallbackQueueImpl	@	react-dom.development.js:11322
flushSyncCallbackQueue	@	react-dom.development.js:11309
scheduleUpdateOnFiber	@	react-dom.development.js:21893
dispatchAction	@	react-dom.development.js:16139
```

**Comment by MubashirullahD** [comment]
We've migrated away from react quill. There were many options, my colleague went to quill directly and used that.

**Comment by VaguelySerious** [comment]
For those of you still struggling with this: I'm maintaining a fork that fixes this issue. Try replacing `react-quill` with the forked [react-quill-new](https://www.npmjs.com/package/react-quill-new) in the interim until `react-quill` is updated. Also follow https://github.com/zenoamaro/react-quill/pull/973 for updates on that.

**Comment by Luminilion** [comment]
@VaguelySerious You are a champ thanks !!

**Comment by HaSungJe** [comment]
@VaguelySerious Thank you !

**Comment by davidfcopozo** [comment]
> For those of you still struggling with this: I'm maintaining a fork that fixes this issue. Try replacing `react-quill` with the forked [react-quill-new](https://www.npmjs.com/package/react-quill-new) in the interim until `react-quill` is updated. Also follow [#973](https://github.com/zenoamaro/react-quill/pull/973) for updates on that.

I tried it but the syntax module seems not to be working, I’ve true Highlight.js Prismjs and nothing works and that’s an important feature for a blog about programming.
